libs/barista-components/chart/src/chart.ts

```diff
@@ -231,11 +231,19 @@ export class DtChart
     }
     if (series instanceof Observable) {
       this._dataSub = series.subscribe((s: DtChartSeries[]) => {
-        this._currentSeries = s;
+        this._currentSeries = s.map(
+          retainSeriesVisibility(this._chartObject?.series),
+        );
         this._update();
       });
     } else {
-      this._currentSeries = series;
+      if (series) {
+        this._currentSeries = series.map(
+          retainSeriesVisibility(this._chartObject?.series),
+        );
+      } else {
+        this._currentSeries = series;
+      }
```

Comment:
NIT: This would reduce duplication and can eliminate an unnecessary variable declaration.
I thought we have the `prefer-conditional-expression` lint rule in place for that... 🤔

```suggestion
      this._currentSeries = !series
          ? series
          : series.map(retainSeriesVisibility(this._chartObject?.series));
```